common/src/main/java/org/wso2/testgrid/common/DashboardSetup.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.common;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.util.StringUtil;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * This class create a database in influxDB and a Data Source in
+ */
+public class DashboardSetup {
+
+    private static final Logger logger = LoggerFactory.getLogger(DashboardSetup.class);
+    private String restUrl = "http://ec2-54-167-98-19.compute-1.amazonaws.com";
+    private String username = "wso2";
+    private String password = "wso2123";
+    private String apikey = "Bearer eyJrIjoiZlZlVDNWMXZFbzdZdGp6R3NMaWlvdmRaRE5sYjFuWXkiLCJuIjoid3NvMiIsImlkIjoxfQ==";
+    private String testplanID;
+
+    public DashboardSetup(String testplanID) {
+        this.testplanID = testplanID;
+    }
+
+    /**
+     * this method will create a DB in influxDB and add a new Data source to grafana
+     */
+    public void initDashboard() {
+        // create influxDB database according to tp_id
+        try {
+            InfluxDB influxDB = InfluxDBFactory.connect(restUrl + ":8086", username, password);
+            String dbName = testplanID;
+            influxDB.createDatabase(dbName);
+            influxDB.close();
+        } catch (AssertionError e) {
+            logger.error(StringUtil.concatStrings("Cannot create a new Database: ", e));
```

Comment:
This can be changed to logger.error(msg, e) without concatanating the throwable to the message string.